### ISSUE TITLE
[Bug] runtime_env doesn't honor env_vars first

### ISSUE THREAD

**Issue by valiantljk** [ray-project/ray#22610]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Clusters

### What happened + What you expected to happen

I set both env_vars and pip in runtime_env, one of the pip install need to access the user-defined env during runtime, but looks like the env is not honored and pip install failed, with errors:
```
File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/util/client/server/proxier.py", line 234, in _create_runtime_env
    "Failed to create runtime_env for Ray client "
RuntimeError: Failed to create runtime_env for Ray client server: Failed to install conda environment /tmp/ray/session_2022-02-23_13-49-17_742804_6/runtime_resources/conda/bda0b0d51deac4d51a7063a37ecbbc2a53c9e840:
raise error
RuntimeError: java home not found, try setting JAVA_HOME
```

### Versions / Dependencies

ray 1.9.0
python 3.7

### Reproduction script

```
import ray,os
runtime_env = {
        "env_vars": {"JAVA_HOME": "/opt/tiger/jdk/jdk1.8"},
       "pip": ["pydoop==2.0.0",],
 }

ray.init("ray://head:port",runtime_env=runtime_env)
print(ray.cluster_resources())
#@ray.remote
#def f():
#  return os.environ.get("JAVA_HOME")

#print(ray.get(f.remote())) # this line works as expected, i.e., JAVA_HOME was set correctly


```

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by grechaw** [comment]
Is this an issue with pydoop installation?  I verified that JAVA_HOME gets set correctly for both Ray 1.9 and 1.10... but what's possible here is that it's not set *at the time that pip installs pydoop*

**Comment by valiantljk** [comment]
Yeah, I can get the var using os.environ (but I guess, this is in the stage of the actual worker process execution); During runtime_env setting up, pip install doesn't see it.

**Comment by Catch-Bull** [comment]
> Is this an issue with pydoop installation? I verified that JAVA_HOME gets set correctly for both Ray 1.9 and 1.10... but what's possible here is that it's not set _at the time that pip installs pydoop_

the `env_vars` is the env of the ray worker process, but we use the command `pip install` to install the python package in the `runtime env agent` process, and `pip install` will run in the child process of the `runtime env agent` process, So env_vars will not take effect on the process which executes pip install command.

but, we can set the env of the `pip install` process by this argument [`env`](https://github.com/ray-project/ray/blob/master/python/ray/_private/runtime_env/pip.py#L237) , do we need to support this on runtime_env? @architkulkarni

**Comment by Catch-Bull** [comment]
So, I think this is not a bug, we just don't support it...

**Comment by Catch-Bull** [comment]
@valiantljk You can try it on the master branch now, this issue should have been resolved

**Comment by architkulkarni** [comment]
Closed by #22730
